R/sfc.R

```diff
@@ -142,8 +142,12 @@ sfg_is_empty = function(x) {
 #' @export
 #"[<-.sfc" = function (x, i, j, value) {
 "[<-.sfc" = function (x, i, value) {
-	if (is.null(value) || inherits(value, "sfg"))
+	if (is.null(value) || inherits(value, "sfg")) {
 		value = list(value)
+	} else if(inherits(value, "vctrs_vctr")) {
+		value = vctrs::vec_cast(value, vctrs::vec_ptype(x))
```

Comment:
Do we need `vec_ptype()` here?

```suggestion
		value = vctrs::vec_cast(value, x)
```